src/transform/FooterContainer.js

```diff
@@ -0,0 +1,91 @@
+import './FooterContainer.css'
+import Polyfill from './Polyfill'
+
+/**
+ * Ensures the 'Read more' section header can always be scrolled to the top of the screen.
+ * @param {!Window} window
+ * @return {void}
+ */
+const updateBottomPaddingToAllowReadMoreToScrollToTop = window => {
+  const div = window.document.getElementById('pagelib_footer_container_ensure_can_scroll_to_top')
+  const currentPadding = parseInt(div.style.paddingBottom, 10) || 0
+  const height = div.clientHeight - currentPadding
+  const newPadding = Math.max(0, window.innerHeight - height)
+  div.style.paddingBottom = `${newPadding}px`
+}
+
+/**
+ * Allows native code to adjust footer container margins without having to worry about
+ * implementation details.
+ * @param {!number} margin
+ * @param {!Document} document
+ * @return {void}
+ */
+const updateLeftAndRightMargin = (margin, document) => {
+  const elements = Polyfill.querySelectorAll(document, `
+    #pagelib_footer_container_menu_heading, 
+    #pagelib_footer_container_readmore, 
+    #pagelib_footer_container_legal
+  `)
+  Array.from(elements)
+    .forEach(element => {
+      element.style.marginLeft = `${margin}px`
+      element.style.marginRight = `${margin}px`
+    })
+  const rightOrLeft = document.querySelector('html').dir === 'rtl' ? 'right' : 'left'
+  Array.from(Polyfill.querySelectorAll(document, '.pagelib_footer_menu_item'))
+    .forEach(element => {
+      element.style.backgroundPosition = `${rightOrLeft} ${margin}px center`
```

Comment:
> Do these need to change dynamically or can these inline styles be replaced with a platform style sheet (no JavaScript needed)?

I think dynamically. 

The footer needs to always use the same side-margin as the article content. But on tablet we adjust the article content side margin (as you can see in the last screenshot below) when the table of contents is hidden.

![screen shot 2017-07-17 at 3 19 33 pm](https://user-images.githubusercontent.com/3143487/28292486-09332fdc-6b04-11e7-8931-32fcc6c55e36.png)

![screen shot 2017-07-17 at 3 19 57 pm](https://user-images.githubusercontent.com/3143487/28292488-0933d2d4-6b04-11e7-8910-dacbafb42baf.png)

![screen shot 2017-07-17 at 3 20 08 pm](https://user-images.githubusercontent.com/3143487/28292490-094790ee-6b04-11e7-8df2-86effd8e17a6.png)

![screen shot 2017-07-17 at 3 21 09 pm](https://user-images.githubusercontent.com/3143487/28292487-0933cf46-6b04-11e7-9180-c934e000891b.png)

![screen shot 2017-07-17 at 3 21 19 pm](https://user-images.githubusercontent.com/3143487/28292485-0931681e-6b04-11e7-8b7d-bff1b3868d77.png)

![screen shot 2017-07-17 at 3 21 30 pm](https://user-images.githubusercontent.com/3143487/28292489-09347946-6b04-11e7-8d57-ade046487cd6.png)